packages/react-native-reanimated/src/platform-specific/RNRenderer.ts

```diff
@@ -1,4 +1,49 @@
+/* eslint-disable camelcase */
 // eslint-disable-next-line @typescript-eslint/ban-ts-comment
 // @ts-nocheck
 'use strict';
-export { default as RNRenderer } from 'react-native/Libraries/Renderer/shims/ReactNative';
+
+import { isFabric } from '../PlatformChecker';
+
+function findHostInstanceFastPath(ref: React.Component) {
+  if (ref.__internalInstanceHandle && ref.__nativeTag && ref._viewConfig) {
+    // This is a native ref to a Fabric component
+    return ref;
+  }
+  if (ref._nativeTag && ref.viewConfig) {
+    // This is a native ref to a Paper component
+    return ref;
+  }
+  return undefined;
+}
+
+function resolveFindHostInstance_DEPRECATED() {
+  if (findHostInstance_DEPRECATED !== undefined) {
+    return;
+  }
+  if (isFabric()) {
+    try {
+      findHostInstance_DEPRECATED =
+        require('react-native/Libraries/Renderer/shims/ReactFabric').findHostInstance_DEPRECATED;
+    } catch (e) {
+      findHostInstance_DEPRECATED = (_ref: unknown) => null;
+    }
+  } else {
+    findHostInstance_DEPRECATED =
+      require('react-native/Libraries/Renderer/shims/ReactNative').findHostInstance_DEPRECATED;
+  }
+}
+
+let findHostInstance_DEPRECATED: (ref: unknown) => void;
+export function findHostInstance(component: React.Component): unknown {
+  // Fast path for native refs
+  const hostInstance = findHostInstanceFastPath(component._componentRef);
+  if (hostInstance !== undefined) {
+    return hostInstance;
+  }
+
+  resolveFindHostInstance_DEPRECATED();
+  return findHostInstance_DEPRECATED(
+    isFabric() ? component : component._componentRef
```

Comment:
Looks like we can skip this `isFabric()` call here and just call a function that uses `findHostInstance_DEPRECATED` under the hood.

Reply:
What do you mean?

Reply:
Let's keep the `resolve` call on each invocation then.